zeroex/order.go

```diff
@@ -74,38 +75,102 @@ const (
 	OSInvalidTakerAssetData
 )
 
+// ContractEventParameters is the parameters of a ContractEvent
+type ContractEventParameters interface {
+	json.Marshaler
+}
+
+// ContractEvent is an event emitted by a smart contract
+type ContractEvent struct {
+	BlockHash  common.Hash
+	TxHash     common.Hash
+	TxIndex    uint
+	LogIndex   uint
+	IsRemoved  bool
+	Address    common.Address
+	Kind       string
+	Parameters ContractEventParameters
+}
+
+// MarshalJSON implements a custom JSON marshaller for the ContractEvent type
+func (c ContractEvent) MarshalJSON() ([]byte, error) {
+	m := map[string]interface{}{
+		"blockHash": c.BlockHash.Hex(),
+		"txHash":    c.TxHash.Hex(),
+		"txIndex":   c.TxIndex,
+		"logIndex":  c.LogIndex,
+		"isRemoved": c.IsRemoved,
+		"address":   c.Address,
+		"kind":      c.Kind,
+	}
+	switch c.Kind {
+	case "ERC20TransferEvent":
```

Comment:
At this point I think you can remove this `switch` statement altogether. `json.Marshal` already iterates each value in the map and calls `MarshalJSON` if possible.